### ISSUE TITLE
Hook exceptions not working on modules' controllers

### ISSUE THREAD

**Issue by daresh** [PrestaShop/PrestaShop#27056]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [x] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

When adding an exception to a module for a hook, it is not taken into account on modules controllers.

### Expected behavior

Exceptions should be taken into account everywhere correctly.

### Steps to reproduce
>Edited by Marwa

1. edit displayHeader for ps_imageslider
2. add an exception for the category and controllers related to the wishlist module
3. check that on the category page there’s no JS/CSS related to the module, but on the wishlist when you visit your list (as a logged customer), JS and CSS are still loaded and they don’t respect the settings 

----------------------------------------------------------------------------

1. Go to Design -> Positions.
2. Set exceptions for some built in module, for example Image Slider, to only be active for index (displayHeader hook), all other pages marked as an exception.
3. Go to some page other than index (category, cms, product) to verify that modules/ps_imageslider/js/responsiveslides.min.js is not present in the source code - OK so far
4. Go to some page that is a module's controller, like module/ps_emailsubscription/subscription to see that modules/ps_imageslider/js/responsiveslides.min.js is present in the source code, even after setting an exception for that controller.

The above is only an example, it seems to be a general issue, cause we also observed it for other modules.

### PrestaShop version(s) where the bug happened

1.7.8.2

### PHP version(s) where the bug happened

7.3.3

**Comment by hibatallahAouadni** [comment]
Hello @daresh 

> 2. Set exceptions for some built in module, for example Image Slider, to only be active for index (displayHeader hook), all other pages marked as an exception.
> 3. Go to some page other than index (category, cms, product) to verify that modules/ps_imageslider/js/responsiveslides.min.js is not present in the source code - OK so far
> 4. Go to some page that is a module's controller, like module/ps_emailsubscription/subscription to see that modules/ps_imageslider/js/responsiveslides.min.js is present in the source code, even after setting an exception for that controller.

Could you please provide a screenhsot or a screen record for these steps :pray: 
Please check and feedback.

Thanks!

**Comment by daresh** [comment]
Sure:
1. Take any module attached to the displayHeader hook, for example Image Slider:
![2021-12-22 20_32_01-Positions • PS177 Dev — Firefox Developer Edition](https://user-images.githubusercontent.com/5658940/147145984-7d2e8193-2c38-40ab-9b4c-8a1a85d5fbb8.png)

2. Set exceptions for it to not be active on pages other than index.
![2021-12-22 20_32_37-Positions _ Edit • PS177 Dev — Firefox Developer Edition](https://user-images.githubusercontent.com/5658940/147146010-a281861b-4f0f-47ba-9fc0-701b5142a6cf.png)

3. Go to a page that is added as a module front controller and see, that the hook is still active, module's JS files are included, it did not follow the exception settings.
![2021-12-22 20_33_28-http___dev greenmou se_ps178_module_ps_emailsubscription_subscription — Firefox ](https://user-images.githubusercontent.com/5658940/147146022-76401925-ea1e-48a5-8d3d-c72083d494ef.png)

Example of a page like this, that should be present in a clean PS installation: module/ps_emailsubscription/subscription

**Comment by marwachelly** [comment]
Hello,

After the steps 1 and 2 I have the following alert

![image](https://user-images.githubusercontent.com/89604350/147582254-b01d1e9a-59a4-47d8-b85d-017f151908ad.png)
After the step 3 I have
![image](https://user-images.githubusercontent.com/89604350/147582435-168275a4-3d39-478c-8514-4a8ab46182d4.png)

Despite our efforts, we were not able to reproduce your issue with the provided information. It seems that your issue is not a PrestaShop core bug but most likely a module problem.

We will close this issue.

Thanks,

**Comment by daresh** [comment]
It's not a module issue, we tested it on a couple of stores, it's a bug in the exceptions mechanism.

Maybe I should show some more screenshots, because you seem to be testing it in an incorrect way.

@marwachelly

**Comment by Lionel-dev** [comment]
I confirm what daresh says... We have the problem on several shops... It's quite annoying...

**Comment by daresh** [comment]
@marwachelly please reopen this issue, otherwise I will need to submit another bug request for it, but that's redundant.

**Comment by kpodemski** [comment]
@marwachelly 

As discussed on Slack, this is a valid report, I was able to reproduce this issue.

**Comment by Lionel-dev** [comment]
@kpodemski personally, I just fixed it on the prestablog module and I think the bug is probably related to it... 

I dumped a bit in the Hook.php class and I realized that "php_self" or "controller_name" was at "null". So I added this in the init function of my controller:

`Context::getContext()->controller->controller_name = 'blog';`

**Comment by kpodemski** [comment]
@Lionel-dev 

Setting `php_self` in the controller might be tricky with some of the third-party SEO modules that generate some stuff based on the `php_self`, overall I remember this issue from years ago, but I forgot to go back to it.

**Comment by Lionel-dev** [comment]
@kpodemski yes I saw while doing my fix thanks :) So I just assigned a controller name and left the php_self as it was :) For the moment I have no side effects
